src/main/kotlin/org/elaastic/questions/assignment/sequence/interaction/InteractionRepository.kt

```diff
@@ -19,10 +19,34 @@
 package org.elaastic.questions.assignment.sequence.interaction
 
 import org.elaastic.questions.assignment.sequence.Sequence
+import org.elaastic.questions.assignment.sequence.interaction.response.Response
+import org.elaastic.questions.directory.User
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
+import org.springframework.data.repository.query.Param
 
 
 interface InteractionRepository : JpaRepository<Interaction, Long> {
 
     fun findAllBySequence(sequence: Sequence): List<Interaction>
+
+    @Query("SELECT DISTINCT r " +
+           "FROM Interaction i " +
+           "INNER JOIN Response r " +
+           "ON i = r.interaction " +
+           "WHERE i.sequence = :sequence " +
+           "AND i.interactionType = :interactionType")
+    fun findAllResponsesBySequenceAndType(sequence: Sequence,
+                                   interactionType: InteractionType): List<Response>
+
+    @Query("SELECT DISTINCT r " +
+           "FROM Interaction i " +
+           "INNER JOIN Response r " +
+           "ON i = r.interaction " +
+           "WHERE i.owner = :owner " +
+           "AND i.sequence = :sequence " +
+           "AND i.interactionType = :interactionType")
+    fun findResponseByOwnerAndSequenceAndType(owner: User,
+                                              sequence: Sequence,
+                                              interactionType: InteractionType): Response?
```

Comment:
why interactionType is a parameter of this method?

Reply:
Read https://github.com/elaastic/elaastic-questions-server/pull/247#discussion_r1634402781